src/rest_parser.cpp

```diff
@@ -654,7 +659,7 @@ Status KFSRestParser::parseInputs(rapidjson::Value& node) {
     }
     requestProto.mutable_inputs()->Clear();
     for (auto& input : node.GetArray()) {
-        auto status = parseInput(input);
+        auto status = parseInput(input, (node.GetArray().Size() == 1));
```

Comment:
parseInput would be capable to deduce if there is only one input itself. Move `(node.GetArray().Size() == 1)` to the parseInput method